default/index.js

```diff
@@ -12,23 +10,36 @@
  *     ---dir1
  *         ---dir2
  *             ---file2
- * The object returned will be:
- * {
- *    my-project: [{
- *                   path: /file1,
- *                   buffer/stream: <file1 contents/stream>
- *                }, {
- *                   path: /dir1/dir2/file2,
- *                   buffer/stream: <file2 contents/stream>
- *                }]
- * }
+ *
+ * Calling defaultProjects.get('my-project', true) will return:
+ *  [{
+ *     path: /file1,
+ *     size: 12345
+ *     stream: <file1 stream>
+ *  }, {
+ *     path: /dir1/dir2/file2,
+ *     size: 12345
+ *     stream: <file2 stream>
+ *  }]
+ *
+ * Calling defaultProjects.get('my-project') will return:
+ *  [{
+ *     path: /file1,
+ *     buffer: <file1 buffer>
+ *  }, {
+ *     path: /dir1/dir2/file2,
+ *     buffer: <file2 buffer>
+ *  }]
  *
  */
 
 var fs = require("fs");
 var Path = require("path");
+var MemoryStream = require("memorystream");
+
+var defaultProjects = {};
 
-function readDirectory(dirName, stream) {
+function readDirectory(dirName) {
   var contents = fs.readdirSync(dirName);
```

Comment:
Because this can throw (all sync fs operations can), I'd suggest you wrap them in `try/catch`, or else your app may go down if it can't do any of this for some reason.